node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -45,30 +45,106 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
+  private val _peerData: mutable.Map[Peer, PeerData] =
     mutable.Map.empty
 
-  def peerServices: Map[Peer, ServiceIdentifier] = _peerServices.toMap
+  def peerData: Map[Peer, PeerData] = _peerData.toMap
+
+  def peers: Vector[Peer] = peerData.keys.toVector
+
+  /** Returns peers by querying each dns seed once. These will be IPv4 addresses. */
+  def getPeersFromDnsSeeds: Vector[Peer] = {
+    val dnsSeeds = nodeAppConfig.network.dnsSeeds
+    val addresses = dnsSeeds
+      .flatMap(seed => {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  def setPeerServices(
-      peer: Peer,
-      serviceIdentifier: ServiceIdentifier): Unit = {
-    _peerServices.put(peer, serviceIdentifier)
-    ()
+  /** Returns peers from hardcoded addresses taken from https://github.com/bitcoin/bitcoin/blob/master/contrib/seeds/nodes_main.txt */
+  def getPeersFromResources: Vector[Peer] = {
+    val source = Source.fromURL(getClass.getResource("/hardcoded-peers.txt"))
+    val addresses = source
+      .getLines()
+      .toVector
+      .filter(nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns all peers stores in db */
+  def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDb]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val filteredAddresses = addresses.filter(
+        nodeAppConfig.torConf.enabled || _.networkId != AddrV2Message.TOR_V3_NETWORK_BYTE)
+      val inetSockets = filteredAddresses.map(a => {
+        NetworkUtil.parseInetSocketAddress(a.address, a.port)
+      })
+      val peers =
+        inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+      peers
+    }
+    peersF
+  }
+
+  def getPeersFromConfig: Vector[Peer] = {
+    val addresses = nodeAppConfig.peers.filter(
+      nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  def getPeers: Future[Vector[Peer]] = {
+    val peersFromConfig = getPeersFromConfig
+    lazy val peersFromDbF = getPeersFromDb
+    val maxConnectedPeers = 1
```

Comment:
I think we had some point talked about propogating this out to a configuration in `bitcoin-s.conf`  and then make it accessible via `NodeAppConfig.maxConnectedPeers`